AzureMonitorAgent/agent.py

```diff
@@ -388,16 +388,22 @@ def enable():
     the settings provided are incomplete or incorrect.
     Note: enable operation times out from WAAgent at 5 minutes
     """
-    global AMAServiceStartCommand, AMAServiceRestartCommand, AMAServiceStatusCommand
 
-    # start the metrics process if its not already running
     public_settings, protected_settings = get_settings()
 
-    if (protected_settings is None or len(protected_settings) == 0) or (public_settings is not None and "proxy" in public_settings and "mode" in public_settings.get("proxy") and public_settings.get("proxy").get("mode") == "application"):
+    # start the metrics watcher if its not already running
+    if ((protected_settings is None or len(protected_settings) == 0) or
+        (public_settings is not None and "proxy" in public_settings and "mode" in public_settings.get("proxy") and public_settings.get("proxy").get("mode") == "application") or
+        (public_settings is not None and public_settings.get("azureMonitorConfiguration") is not None and public_settings.get("azureMonitorConfiguration").get("ensure") == True)):
```

Comment:
Do you want to move "azureMomitorConfiguration" to be a const? I see it repeated , same for ""azuremonitoragent"

Reply:
Looks good. I can sign off assuming upgrade with old settings and brand new install, both the paths were tested?

Reply:
+1

Reply:
I tested it as follows
- next-gen geneva AND azure monitor config
- next-gen geneva config
- next-gen azure monitor config
- next-gen azure monitor config with parameters (e.g. managed identity)
- legacy geneva config
- legacy autoconfig
- legacy azure monitor config
- mixed legacy and next-gen (results in error as expected)